scipy/linalg/decomp.py

```diff
@@ -797,22 +838,232 @@ def eigvals_banded(a_band, lower=False, overwrite_a_band=False,
+    tol : float
+        The absolute tolerance to which each eigenvalue is required
+        (only used when 'stebz' is the `lapack_driver`).
+        An eigenvalue (or cluster) is considered to have converged if it
+        lies in an interval of this width. If <= 0. (default),
+        the value ``eps*|a|`` is used where eps is the machine precision,
+        and ``|a|`` is the 1-norm of the matrix ``a``.
+    check_finite : bool, optional
+        Whether to check that the input matrix contains only finite numbers.
+        Disabling may give a performance gain, but may result in problems
+        (crashes, non-termination) if the inputs do contain infinities or NaNs.
+    lapack_driver : str
+        LAPACK function to use. When 'auto' (default), it will use 'stemr'
+        if ``select='a'`` and 'stebz' otherwise. When 'stebz' is used to find
+        the eigenvalues and ``eigvals_only=False``, then a second LAPACK call
+        (to ``?STEIN``) is used to find the corresponding eigenvectors.
+        'sterf' can only be used when ``eigvals_only=True, select='a'``.
+
+    Returns
+    -------
+    w : (M,) ndarray
+        The eigenvalues, in ascending order, each repeated according to its
+        multiplicity.
+    v : (M, M) ndarray
+        The normalized eigenvector corresponding to the eigenvalue ``w[i]`` is
+        the column ``v[:,i]``.
+
+    Raises
+    ------
+    LinAlgError
+        If eigenvalue computation does not converge.
+
+    See Also
+    --------
+    eigvals_tridiagonal : eigenvalues of symmetric/Hermitian tridiagonal
+        matrices
+    eig : eigenvalues and right eigenvectors for non-symmetric arrays
+    eigh : eigenvalues and right eigenvectors for symmetric/Hermitian arrays
+    eig_banded : eigenvalues and right eigenvectors for symmetric/Hermitian
+        band matrices
+
+    Notes
+    -----
+    This function makes use of LAPACK ``S/DSTEMR`` routines.
+    """
+    d = _asarray_validated(d, check_finite=check_finite)
+    e = _asarray_validated(e, check_finite=check_finite)
+    for check in (d, e):
+        if check.ndim != 1:
+            raise ValueError('expected one-dimensional array')
+        if check.dtype.char in 'GFD':  # complex
+            raise TypeError('Only real arrays currently supported')
+    assert d.size > 0
+    if d.size != e.size + 1:
+        raise ValueError('d (%s) must have one more element than e (%s)'
+                         % (d.size, e.size))
+    select, vl, vu, il, iu, _ = _check_select(
+        select, select_range, 0, d.size)
+    if not isinstance(lapack_driver, string_types):
+        raise TypeError('lapack_driver must be str')
+    drivers = ('auto', 'stemr', 'sterf', 'stebz')
+    if lapack_driver not in drivers:
+        raise ValueError('lapack_driver must be one of %s, got %s'
+                         % (drivers, lapack_driver))
+    if lapack_driver == 'auto':
+        lapack_driver = 'stebz' if select == 0 else 'stemr'
```

Comment:
This line uses the wrong case for `'stebz'`. When I do `..., select='i', select_range=(n-10, n-1), ...` then the numerical select code ends up being 2.